src/coreclr/vm/threads.inl

```diff
@@ -27,7 +27,7 @@ EXTERN_C UINT32 _tls_index;
 #endif
 
 #ifdef _MSC_VER
-__declspec(selectany) thread_local
+__declspec(selectany) __declspec(thread)
```

Comment:
This is the only perf critical instance in this PR that needs special treatment. I think the rest can stay as `thread_local`.